client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -14,31 +15,150 @@ const EditProfilePage = () => {
 
-  const [formData, setFormData] = useState({
+  type FormDataType = {
+    email: string;
+    nickName: string;
+    cohort: string;
+    personalBio: string;
+    careerInformation: {
+      currentPosition: {
+        title: string;
+        company: string;
+      };
+    };
+    availabilityForNetworking: boolean;
+    linkedInProfile: string;
+    gitHubProfile: string;
+    socialMediaLinks: {
+      twitter: string;
+      blog: string;
+    };
+    skills: string[];
+    specializations: string[];
+  };
+  
+  const [formData, setFormData] = useState<FormDataType>({
     email: '',
+    nickName: '',
+    cohort: '',
     personalBio: '',
+     careerInformation: {
+      currentPosition: {
+        title: '',
+        company: '',
+      },
+    },
+    availabilityForNetworking: false,
+    linkedInProfile: '',
+    gitHubProfile: '',
+    socialMediaLinks: {
+      twitter: '',
+      blog: '',
+    },
+    skills: [] as string[],
+    specializations: [] as string[],
   });
 
+  const [skillInput, setSkillInput] = useState('');
+
   const [file, setFile] = useState<File | null>(null);
 
   useEffect(() => {
     if (userID) dispatch(fetchUserProfile(userID as string));
-  }, [dispatch]);
+  }, [dispatch, userID]);
+
   useEffect(() => {
     if (profile) {
       setFormData({
         email: profile.email || '',
+        nickName: profile.nickName || '',
+        cohort: profile.cohort || '',
         personalBio: profile.personalBio || '',
+        careerInformation: {
+          currentPosition: {
+            title: profile.careerInformation?.currentPosition?.title || '',
+            company: profile.careerInformation?.currentPosition?.company || '',
+          },
+        },
+        availabilityForNetworking: profile.availabilityForNetworking || false,
+        linkedInProfile: profile.linkedInProfile || '',
+        gitHubProfile: profile.gitHubProfile || '',
+        socialMediaLinks: {
+          twitter: profile.socialMediaLinks?.twitter || '',
+          blog: profile.socialMediaLinks?.blog || '',
+        },
+        skills: profile.skills || [],
+        specializations: profile.specializations || [], 
       });
     }
   }, [profile]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const { name, value } = e.target;
-    setFormData((prevFormData) => ({
-      ...prevFormData,
-      [name]: value,
-    }));
+    const keys = name.split('.');
+    if (keys.length === 1) {
+      setFormData((prevFormData) => ({
+        ...prevFormData,
+        [name]: value,
+      }));
+    } else {
+      setFormData((prevFormData) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const newFormData: any = { ...prevFormData }; 
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let current: any = newFormData;
```

Comment:
It can be tricky working with nested data types. A common design pattern is to construct the nested structure as late as possible. In this case, you could refactor `formData` to only be 1 level deep, and in your event handler, add the necessary level of nesting.

Reply:
This is a really clever way to dynamically handle the different levels of nesting! The downside it def seems to require using an any type or doing some typing gymnastics which might make it more trouble than it's worth. I agree with Jimmy, it makes more sense to me to unnest the formData state and then "re-nest" it in the handleSubmit.

I do think there's a conversation to be had about why we're nesting these fields in the schema and what, if anything are we gaining from that vs. having them unnested

Reply:
Great idea to un-nest form data and then re-nest in handleSubmit @JimmyTran48 @seantokuzo ! I've just followed your suggestion and it looks like thats gonna work. Thank you!!! 

Running through some final testing now before pushing up.